bee-node/bee-node/src/main.rs

```diff
@@ -1,31 +1,112 @@
 use bee_node::{
     banner::print_logo_and_version,
     cli::NodeCliArgs,
     config::{NodeConfigBuilder, DEFAULT_NODE_CONFIG_PATH},
 };
 
-#[tokio::main(flavor = "multi_thread")]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let cli = NodeCliArgs::new();
+use backstage::core::{
+    AbortableUnboundedChannel, Actor, ActorError, ActorResult, EolEvent, ReportEvent, Rt, Runtime, ScopeId, Service,
+    StreamExt, SupHandle,
+};
+
+#[derive(Default)]
+struct BeeSupervisor {}
+
+impl BeeSupervisor {
+    fn new() -> Self {
+        Self::default()
+    }
+}
+
+enum BeeSupervisorEvent {
+    Eol,
+    Report,
+}
+
+impl<T> EolEvent<T> for BeeSupervisorEvent {
+    fn eol_event(_scope: ScopeId, _service: Service, _actor: T, _r: ActorResult<()>) -> Self {
+        Self::Eol
+    }
+}
+
+impl<T> ReportEvent<T> for BeeSupervisorEvent {
+    fn report_event(_scope: ScopeId, _service: Service) -> Self {
+        Self::Report
+    }
+}
+
+#[async_trait::async_trait]
+impl<S: SupHandle<Self>> Actor<S> for BeeSupervisor {
+    type Data = ();
 
-    let config = match NodeConfigBuilder::from_file(cli.config().unwrap_or(DEFAULT_NODE_CONFIG_PATH)) {
-        Ok(builder) => builder.with_cli_args(cli.clone()).finish(),
-        Err(e) => panic!("failed to create the node config builder: {}", e),
-    };
+    type Channel = AbortableUnboundedChannel<BeeSupervisorEvent>;
 
-    if let Err(e) = logger_init(config.logger) {
-        panic!("failed to initialise the logger: {}", e);
+    async fn init(&mut self, _rt: &mut Rt<Self, S>) -> ActorResult<Self::Data> {
+        Ok(())
     }
 
-    print_logo_and_version();
+    async fn run(&mut self, rt: &mut Rt<Self, S>, _data: Self::Data) -> ActorResult<()> {
+        let cli = NodeCliArgs::new();
 
-    if cli.version() {
-        return Ok(());
+        let config = NodeConfigBuilder::from_file(cli.config().unwrap_or(DEFAULT_NODE_CONFIG_PATH))
+            .map_err(ActorError::aborted)?
+            .with_cli_args(cli.clone())
+            .finish();
+
+        logger_init(config.logger).map_err(ActorError::aborted)?;
+
+        print_logo_and_version();
+
+        if cli.version() {
+            return Ok(());
+        }
+
+        let port = cli.port();
+        let identity = cli.identity();
+
+        let local_id = if let Some(identity) = identity {
+            LocalIdentity::from_bs58_secret_key_str(identity)
+        } else {
+            LocalIdentity::new()
+        };
+
+        log::info!("node local id: {:?}", local_id);
+
+        let config = Config::new(port, local_id, "./peers.yaml").map_err(ActorError::exit)?;
+
+        rt.add_resource(config).await;
+
+        rt.start(Some("network".into()), NetworkWorker::new()).await.unwrap();
+
+        while let Some(event) = rt.inbox_mut().next().await {
+            match event {
+                BeeSupervisorEvent::Eol | BeeSupervisorEvent::Report => {
+                    // FIXME: handle network status report events.
```

Comment:
Does this need implementing?

Reply:
@pvdrz Wdyt?

Reply:
Ideally, both of them should be merged into one variant.